### DISCUSSION TITLE
GitHub Actions: get the PR ID that started it?

### DISCUSSION THREAD

**Discussion by mrcasals** [community/community#26128] [Actions]
<p>Hi all!</p>
<p>I’m starting with GitHub Actions, and I wanted to send a Slack notification when a PR review was submitted. This is the workflow I’m using:</p>
<pre><code>workflow "Notify PR reviews" {
  on = "pull_request_review"
  resolves = ["GitHub Action for Slack"]
}

 action "On review submission" {
  uses = "actions/bin/filter@c6471707d308175c57dfe91963406ef205837dbd"
  args = "action submitted"
}

 action "GitHub Action for Slack" {
  uses = "Ilshidur/action-slack@5b3a58f5e0ff655ca9c17a22516efdf9d0de36bf"
  needs = ["On review submission"]
  secrets = ["SLACK_WEBHOOK"]
  args = "A new PR review was submitted!"
}
</code></pre>
<p>I checked the available <a href="https://developer.github.com/actions/creating-github-actions/accessing-the-runtime-environment/#environment-variables" rel="nofollow noopener">environment variables</a>, but I can’t seem to find any that returns this info. Am I missing something?</p>
<p>Thanks!</p>

**Comment by nickvanw** [comment, marked as answer]
<p>Hi <a class="mention" href="/mrcasals">@mrcasals</a>!</p>
<p>You should be able to get this information by looking at the file located at the environment variable GITHUB_EVENT_PATH, you can find more information on that here: <a href="https://developer.github.com/actions/creating-github-actions/accessing-the-runtime-environment/#environment-variables" rel="nofollow noopener">https://developer.github.com/actions/creating-github-actions/accessing-the-runtime-environment/#environment-variables</a></p>
<p>This should contain information about the Pull Request that has triggered your action.</p>
<p>If I’m misreading anything, let me know!</p>

**Comment by abinoda** [comment]
<p><span class="mention">@mrcasalsCan</span> you post your use case <a href="https://github.com/pullreminders/slack-action/issues/1" rel="nofollow noopener">here</a>?</p>
<p>I’m working on creating example workflows similar to what you’re describing.</p>

**Comment by mrcasals** [comment]
<p>Done, thanks!</p>
<p>Maybe it’s better if we use an action to set some env variables based on the <code>GITHUB_EVENT_PATH</code> contents, and use your Slack image to send the contents of those variables, as the docs say that “any action can create […] environment variables”:</p>
<p><a href="https://developer.github.com/actions/creating-github-actions/accessing-the-runtime-environment/#environment-variables" rel="nofollow noopener">https://developer.github.com/actions/creating-github-actions/accessing-the-runtime-environment/#environment-variables</a></p>

**Comment by mrcasals** [comment]
<p>Hi <a class="mention" href="/nickvanw">@nickvanw</a>! Thanks for your reply!</p>
<p>I didn’t realize the <code>GITHUB_EVENT_PATH</code> pointed to a file in the filesystem, guess I read that section diagonally. Thanks! I don’t understand what the “payload of the event” is, though. How can I debug what this file contains? I might be missing something :frowning:</p>
<p>I see that “any action can create […] environment variables”, as per the link you posted. Could we create an action that sets env variables based on the contents of the file, and then use those variables as args for the action that sends the message? </p>

**Comment by abinoda** [comment]
<p><span class="mention">@mrcasalsI</span> think we’re thinking along the same lines. I responded to your comment on that issue. If you’d like, maybe close this ticket here and we can continue collaborating on the issue thread?</p>

**Comment by nickvanw** [comment]
<p>Hey <a class="mention" href="/mrcasals">@mrcasals</a>!</p>
<p>It looks like you’re already working with <a class="mention" href="/abinoda">@abinoda</a>, but if you look inside of that file you’ll find that it matches the same format that’s documented for our webhooks (<a href="https://developer.github.com/v3/activity/events/types/" rel="nofollow noopener">https://developer.github.com/v3/activity/events/types/</a>) - so if you’re looking at PRs, for example, it should follow the same format here: <a href="https://developer.github.com/v3/activity/events/types/#pullrequestevent" rel="nofollow noopener">https://developer.github.com/v3/activity/events/types/#pullrequestevent</a></p>
<p>From there, you should be able to parse it as JSON and extract the data yourself. It looks like <a class="mention" href="/abinoda">@abinoda</a> is working on a generic way to extract these, but you should be able to use your language of choice to parse the information you need from that file. </p>

**Comment by haptikrajasashtikar** [comment]
<p>I have just started experimenting with this…<br>
Could you please point me to a sample of this?</p>
<p>action “GitHub Action for Slack” {<br>
uses = “Ilshidur/action-slack@f37693b4e0589604815219454efd5cb9b404fb85”<br>
secrets = [“SLACK_WEBHOOK”]<br>
args =<br>
}</p>
<p>I want a link to the PR in the args</p>
